### DISCUSSION TITLE
Comprehensive test failure

### DISCUSSION THREAD

**Discussion by shinysyk** [neuropoly/axondeepseg#690] [Help needed]
Hi. Thanks for developing and sharing this wonderful tool. During the comprehensive test, one failed with the following error message.

AttributeError: Can't get attribute '_unpickle_block' on <module 'pandas._libs.internals' from '/usr/local/lib/python3.8/dist-packages/pandas/_libs/internals.cpython-38-x86_64-linux-gnu.so'>

I suspect that this is because the pandas version you used for axondeepseg is not compatible with the one on my linux (I tried 1.0.5, 1.2.5, and 1.3.4). 

Please help me with this issue. Thanks for your help in advance.

**Comment by mathieuboudreau** [comment]
Hi @shinysyk !

Package conflicts like these are why we encourage the use of virtual environments; these create isolated instances for you to install the exact package requirements for different projects you might be working on.

We have instructions on how to install miniconda and then create a virtual environment using it in our documentation:

* Installing Miniconda (not necessary if you already have miniconda or anaconda installed): https://axondeepseg.readthedocs.io/en/latest/documentation.html#miniconda
* Creating the ADS conda environment: https://axondeepseg.readthedocs.io/en/latest/documentation.html#install-axondeepseg

Don't forget the final step of activating the environment.

Please let me know if this resolves your issue!

**Reply by shinysyk** [reply]
Hi. Thanks for your swift reply. I installed miniconda and created a virtual environment following the step-by-step instructions at https://axondeepseg.readthedocs.io/en/latest/documentation.html#installation. Under this condition, I encountered this problem. One thing that I remember is that I have my own PythonPATH defined in my bash. I remember that miniconda said that this can cause a weird behavior. Can it cause this conflict? If so, how can I solve the problem?

**Comment by hermancollin** [comment]
Hi @shinysyk.
My version of pandas is 1.4.3. You mentioned you tried 1.0.5, 1.2.5 and 1.3.4. Would it be possible to use the same version as me? Also, note that before launching the comprehensive tests again, you can confirm you are using the expected version with `pip list | grep pandas` which will output the version you have installed.

For the dev team:
I suspect the cause of his problem is that a certain file was saved using a given version of pandas, and from digging around, they change the API for their internal library often. This means that the function that pickled the file is not backwards-compatible with the function that he has on his system. If this is the case, we will need to pin the pandas version.

**Reply by shinysyk** [reply]
Hi. Using pandas 1.4.3 solved the problem. Thanks a lot for your help!!!
